src/App.tsx

```diff
@@ -1,61 +1,136 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUserByName(userName: string): User | null {
+  return usersFromServer.find(user => (user.name === userName)) || null;
+}
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+const prepTodos: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: getUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [todos, setTodos] = useState(prepTodos);
+
+  const [shouldErrorOnUserSelect, setErrorOnUserSelect] = useState(false);
+  const [shouldErrorOnTitleInput, setErrorOnTitleInput] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setErrorOnTitleInput(false);
```

Comment:
Do you need to trigger this function if isErrorOnTitleInput has already been `false` ? You should add a check and trigger the change when `shouldErrorOnTitleInput` is `true`